esp-hal-common/src/soc/esp32s3/psram.rs

```diff
@@ -727,3 +740,1023 @@ pub(crate) mod utils {
+        }
+
+        fn set_rsvd1_2(&mut self, value: u8) {
+            self.mr2 &= !(0b11 << 5);
+            self.mr2 |= (value & 0b11) << 5;
+        }
+
+        fn gb(&self) -> u8 {
+            (self.mr2 >> 7) & 0b1
+        }
+
+        fn set_gb(&mut self, value: u8) {
+            self.mr2 &= !(0b1 << 7);
+            self.mr2 |= (value & 0b1) << 7;
+        }
+
+        fn rsvd3_7(&self) -> u8 {
+            self.mr3 & 0b11111
+        }
+
+        fn set_rsvd3_7(&mut self, value: u8) {
+            self.mr3 &= !0b11111;
+            self.mr3 |= value & 0b11111;
+        }
+
+        fn srf(&self) -> u8 {
+            (self.mr3 >> 5) & 0b1
+        }
+
+        fn set_srf(&mut self, value: u8) {
+            self.mr3 &= !(0b1 << 5);
+            self.mr3 |= (value & 0b1) << 5;
+        }
+
+        fn vcc(&self) -> u8 {
+            (self.mr3 >> 6) & 0b1
+        }
+
+        fn set_vcc(&mut self, value: u8) {
+            self.mr3 &= !(0b1 << 6);
+            self.mr3 |= (value & 0b1) << 6;
+        }
+
+        fn rsvd0(&self) -> u8 {
+            (self.mr3 >> 7) & 0b1
+        }
+
+        fn set_rsvd0(&mut self, value: u8) {
+            self.mr3 &= !(0b1 << 7);
+            self.mr3 |= (value & 0b1) << 7;
+        }
+
+        fn pasr(&self) -> u8 {
+            self.mr4 & 0b111
+        }
+
+        fn set_pasr(&mut self, value: u8) {
+            self.mr4 &= !0b111;
+            self.mr4 |= value & 0b111;
+        }
+
+        fn rf(&self) -> u8 {
+            (self.mr4 >> 3) & 0b1
+        }
+
+        fn set_rf(&mut self, value: u8) {
+            self.mr4 &= !(0b1 << 3);
+            self.mr4 |= (value & 0b1) << 3;
+        }
+
+        fn rsvd3(&self) -> u8 {
+            (self.mr4 >> 4) & 0b1
+        }
+
+        fn set_rsvd3(&mut self, value: u8) {
+            self.mr4 &= !(0b1 << 4);
+            self.mr4 |= (value & 0b1) << 4;
+        }
+
+        fn wr_latency(&self) -> u8 {
+            (self.mr4 >> 5) & 0b111
+        }
+
+        fn set_wr_latency(&mut self, value: u8) {
+            self.mr4 &= !(0b111 << 5);
+            self.mr4 |= (value & 0b111) << 5;
+        }
+
+        fn bl(&self) -> u8 {
+            self.mr8 & 0b11
+        }
+
+        fn set_bl(&mut self, value: u8) {
+            self.mr8 &= !0b11;
+            self.mr8 |= value & 0b11;
+        }
+
+        fn bt(&self) -> u8 {
+            (self.mr8 >> 2) & 0b1
+        }
+
+        fn set_bt(&mut self, value: u8) {
+            self.mr8 &= !(0b1 << 2);
+            self.mr8 |= (value & 0b1) << 2;
+        }
+
+        fn rsvd0_4(&self) -> u8 {
+            (self.mr8 >> 3) & 0b11111
+        }
+
+        fn set_rsvd0_4(&mut self, value: u8) {
+            self.mr8 &= !(0b11111 << 3);
+            self.mr8 |= (value & 0b11111) << 3;
+        }
+    }
+
+    #[ram]
+    pub(crate) fn psram_init() {
+        mspi_pin_init();
+        init_psram_pins();
+        set_psram_cs_timing();
+        // s_configure_psram_ecc();
```

Comment:
Is this line needed?

Reply:
it's meant as a reminder when we want to add ECC support later